wpimath/src/main/native/include/frc/trajectory/TrajectoryGenerator.h

```diff
@@ -82,7 +82,7 @@ class TrajectoryGenerator {
    * @param config    The configuration for the trajectory.
    * @return The generated trajectory.
    */
-  static Trajectory GenerateTrajectory(const std::vector<Pose2d>& waypoints,
+  static Trajectory GenerateTrajectory(std::vector<Pose2d> waypoints,
```

Comment:
Why was this changed to pass-by-value?

Reply:
The poses need to be flipped if the trajectory is reversed. The Java code creates a completely new vector, copies the waypoints over, and then flips them. To avoid writing all that extra code, I can simply take the vector by value and not worry about having to create a copy.

Changing the code from const-reference to value *shouldn't* be breaking, although if there's some corner case, please let me know.

Reply:
This is also why we take in the control vector list here (https://github.com/wpilibsuite/allwpilib/blob/master/wpimath/src/main/native/cpp/trajectory/TrajectoryGenerator.cpp#L79) by value.

Reply:
Can't you just make a temporary copy internally?